ui/signals.py

```diff
@@ -1,17 +1,63 @@
 """
 ui model signals
 """
-from django.db.models.signals import pre_delete
+from django.conf import settings
+from django.db.models.signals import pre_delete, post_save
 from django.dispatch import receiver
 
-from ui.models import VideoFile, VideoThumbnail, VideoSubtitle
+from cloudsync.tasks import upload_youtube_video, remove_youtube_video, remove_youtube_caption
+from ui.models import VideoFile, VideoThumbnail, VideoSubtitle, Video, YouTubeVideo
+
+# pylint: disable=unused-argument
 
 
 @receiver(pre_delete, sender=VideoFile)
 @receiver(pre_delete, sender=VideoThumbnail)
 @receiver(pre_delete, sender=VideoSubtitle)
-def delete_s3_files(sender, **kwargs):  # pylint:disable=unused-argument
+def delete_s3_files(sender, **kwargs):
     """
     Make sure S3 files are deleted along with associated video file/thumbnail object
     """
     kwargs['instance'].delete_from_s3()
+
+
+@receiver(pre_delete, sender=VideoSubtitle)
+def update_video_permissions(sender, **kwargs):
+    """
+    Remove public video permissions if the subtitle is about to be deleted and no other subtitles exist.
+    Otherwise just delete the subtitle from Youtube.
+    """
+    if settings.ENABLE_VIDEO_PERMISSIONS:
+        video = kwargs['instance'].video
+        if video.is_public:
```

Comment:
Can you add a test that we don't attempt to update video permissions if the video is private?

Reply:
OK

Reply:
Added to `test_youtube_video_permissions_signal`